presto-cache/src/main/java/com/facebook/presto/cache/alluxio/AlluxioCachingClientFileSystem.java

```diff
@@ -0,0 +1,254 @@
+    }
+
+    @Override
+    public void mount(AlluxioURI alluxioURI1, AlluxioURI alluxioURI2, MountPOptions options)
+    {
+        throw new UnsupportedOperationException("Mount is not supported.");
+    }
+
+    @Override
+    public void updateMount(AlluxioURI alluxioURI, MountPOptions options)
+    {
+        throw new UnsupportedOperationException("UpdateMount is not supported.");
+    }
+
+    @Override
+    public Map<String, MountPointInfo> getMountTable()
+    {
+        throw new UnsupportedOperationException("GetMountTable is not supported.");
+    }
+
+    @Override
+    public List<SyncPointInfo> getSyncPathList()
+    {
+        throw new UnsupportedOperationException("GetSyncPathList is not supported.");
+    }
+
+    @Override
+    public void persist(AlluxioURI alluxioURI, ScheduleAsyncPersistencePOptions options)
+    {
+        throw new UnsupportedOperationException("Persist is not supported.");
+    }
+
+    @Override
+    public void rename(AlluxioURI source, AlluxioURI destination, RenamePOptions options)
+    {
+        throw new UnsupportedOperationException("ReverseResolve is not supported.");
+    }
+
+    @Override
+    public AlluxioURI reverseResolve(AlluxioURI alluxioURI)
+    {
+        throw new UnsupportedOperationException("ReverseResolve is not supported.");
+    }
+
+    @Override
+    public void setAcl(AlluxioURI alluxioURI, SetAclAction action, List<AclEntry> entries, SetAclPOptions options)
+    {
+        throw new UnsupportedOperationException("SetAcl is not supported.");
+    }
+
+    @Override
+    public void startSync(AlluxioURI alluxioURI)
+    {
+        throw new UnsupportedOperationException("StartSync is not supported.");
+    }
+
+    @Override
+    public void stopSync(AlluxioURI alluxioURI)
+    {
+        throw new UnsupportedOperationException("StopSync is not supported.");
+    }
+
+    @Override
+    public void setAttribute(AlluxioURI alluxioURI, SetAttributePOptions options)
+    {
+        throw new UnsupportedOperationException("SetAttribute is not supported.");
+    }
+
+    @Override
+    public void unmount(AlluxioURI alluxioURI, UnmountPOptions options)
+    {
+        throw new UnsupportedOperationException("unmount is not supported.");
+    }
+
+    @Override
+    public void close()
+            throws IOException
+    {
+        fileSystem.close();
+    }
+
+    private Path toPath(AlluxioURI alluxioURI)
+    {
+        return new Path(alluxioURI.toString());
+    }
+
+    private URIStatus toUriStatus(FileStatus status)
+    {
+        //FilePath is a unique identifier for a file, however it can be a long string
+        //hence using md5 hash of the file path as the identifier in the cache.
+        FileInfo info = new FileInfo();
+        info.setFileIdentifier(md5().hashString(status.getPath().toString(), UTF_8).toString());
+        info.setLength(status.getLen()).setPath(status.getPath().toString());
```

Comment:
`.setPath(status.getPath().toUri().getPath())`  
FileInfo.path should be relative path without scheme

Reply:
Presto filesystem libraries expect the complete path for operations, they don't work with relative path.

Reply:
this will fail if connecting with other file system (like when connecting to Alluxio)

Reply:
@apc999 Interesting point. 
I verified it for presto open source(couple of different filesystems) and facebook. It does require full path.
If to connect to Alluxio does require relative path, let's discuss that offline what can be done to make that work.

Reply:
it is good. I could query Alluxio successfully.

Reply:
Does that mean the PR is good?